domain/consensus/constructors.go

```diff
@@ -0,0 +1,6 @@
+package consensus
+
+import "github.com/kaspanet/kaspad/domain/consensus/model"
+
+// GHOSTDAGConstructor is the function signature for a constructor of a type implementing model.GHOSTDAGManager
+type GHOSTDAGConstructor func(model.DBReader, model.DAGTopologyManager, model.GHOSTDAGDataStore, model.BlockHeaderStore, model.KType) model.GHOSTDAGManager
```

Comment:
Please rename this (and instance variables) to GHOSTDAGManagerConstructor, since we also have GHOSTDAGDataStore.